iam/api-client iam.snippets.ServiceAccountTests.stage1_testServiceAccountCreate

Found in #3602 
```
------------------------------------------------------------
- testing iam/api-client
------------------------------------------------------------
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.217 s <<< FAILURE! - in iam.snippets.ServiceAccountTests
[ERROR] iam.snippets.ServiceAccountTests.stage1_testServiceAccountCreate  Time elapsed: 0.532 s  <<< FAILURE!
java.lang.AssertionError:

Expected: a string containing "Created service account: your-service-account-name"
     but: was "Unable to create service account:
com.google.api.client.googleapis.json.GoogleJsonResponseException: 409 Conflict
POST https://iam.googleapis.com/v1/projects/java-docs-samples-testing/serviceAccounts
{
  "code" : 409,
  "errors" : [ {
    "domain" : "global",
    "message" : "Service account your-service-account-name already exists within project projects/java-docs-samples-testing.",
    "reason" : "alreadyExists"
  } ],
  "message" : "Service account your-service-account-name already exists within project projects/java-docs-samples-testing.",
  "status" : "ALREADY_EXISTS"
}
"
```
assigning to @averikitsch as I think her team might be doing IAM things - feel free to reassign and I'm also trying to find the right person

--- a/iam/api-client/src/main/java/iam/snippets/CreateServiceAccount.java
+++ b/iam/api-client/src/main/java/iam/snippets/CreateServiceAccount.java
@@ -31,8 +31,9 @@ import java.util.Collections;
 public class CreateServiceAccount {
 
   // Creates a service account.
-  public static void createServiceAccount(String projectId) {
+  public static void createServiceAccount(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -46,7 +47,7 @@ public class CreateServiceAccount {
       ServiceAccount serviceAccount = new ServiceAccount();
       serviceAccount.setDisplayName("your-display-name");
       CreateServiceAccountRequest request = new CreateServiceAccountRequest();
-      request.setAccountId("your-service-account-name");
+      request.setAccountId(serviceAccountName);
       request.setServiceAccount(serviceAccount);
 
       serviceAccount =

--- a/iam/api-client/src/main/java/iam/snippets/CreateServiceAccountKey.java
+++ b/iam/api-client/src/main/java/iam/snippets/CreateServiceAccountKey.java
@@ -31,8 +31,9 @@ import java.util.Collections;
 public class CreateServiceAccountKey {
 
   // Creates a key for a service account.
-  public static void createKey(String projectId) {
+  public static void createKey(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -42,6 +43,7 @@ public class CreateServiceAccountKey {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       ServiceAccountKey key =
           service
@@ -49,9 +51,7 @@ public class CreateServiceAccountKey {
               .serviceAccounts()
               .keys()
               .create(
-                  "projects/-/serviceAccounts/your-service-account-name@"
-                      + projectId
-                      + ".iam.gserviceaccount.com",
+                  "projects/-/serviceAccounts/" + serviceAccountEmail,
                   new CreateServiceAccountKeyRequest())
               .execute();
 

--- a/iam/api-client/src/main/java/iam/snippets/DeleteServiceAccount.java
+++ b/iam/api-client/src/main/java/iam/snippets/DeleteServiceAccount.java
@@ -29,8 +29,9 @@ import java.util.Collections;
 public class DeleteServiceAccount {
 
   // Deletes a service account.
-  public static void deleteServiceAccount(String projectId) {
+  public static void deleteServiceAccount(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -40,22 +41,15 @@ public class DeleteServiceAccount {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       service
           .projects()
           .serviceAccounts()
-          .delete(
-              "projects/-/serviceAccounts/"
-                  + "your-service-account-name@"
-                  + projectId
-                  + ".iam.gserviceaccount.com")
+          .delete("projects/-/serviceAccounts/" + serviceAccountEmail)
           .execute();
 
-      System.out.println(
-          "Deleted service account: "
-              + "your-service-account-name@"
-              + projectId
-              + ".iam.gserviceaccount.com");
+      System.out.println("Deleted service account: " + serviceAccountEmail);
     } catch (IOException e) {
       System.out.println("Unable to delete service account: \n" + e.toString());
     }

--- a/iam/api-client/src/main/java/iam/snippets/DeleteServiceAccountKey.java
+++ b/iam/api-client/src/main/java/iam/snippets/DeleteServiceAccountKey.java
@@ -31,8 +31,9 @@ import java.util.List;
 public class DeleteServiceAccountKey {
 
   // Deletes a service account key.
-  public static void deleteKey(String projectId) {
+  public static void deleteKey(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -42,6 +43,7 @@ public class DeleteServiceAccountKey {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       // First, get the name of the key using List() or Get()
       List<ServiceAccountKey> keys =
@@ -49,11 +51,7 @@ public class DeleteServiceAccountKey {
               .projects()
               .serviceAccounts()
               .keys()
-              .list(
-                  "projects/-/serviceAccounts/"
-                      + "your-service-account-name@"
-                      + projectId
-                      + ".iam.gserviceaccount.com")
+              .list("projects/-/serviceAccounts/" + serviceAccountEmail)
               .execute()
               .getKeys();
       String keyToDelete = keys.get(0).getName();

--- a/iam/api-client/src/main/java/iam/snippets/DisableServiceAccount.java
+++ b/iam/api-client/src/main/java/iam/snippets/DisableServiceAccount.java
@@ -30,8 +30,9 @@ import java.util.Collections;
 public class DisableServiceAccount {
 
   // Disables a service account.
-  public static void disableServiceAccount(String projectId) {
+  public static void disableServiceAccount(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -41,24 +42,16 @@ public class DisableServiceAccount {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       DisableServiceAccountRequest request = new DisableServiceAccountRequest();
       service
           .projects()
           .serviceAccounts()
-          .disable(
-              "projects/-/serviceAccounts/"
-                  + "your-service-account-name@"
-                  + projectId
-                  + ".iam.gserviceaccount.com",
-              request)
+          .disable("projects/-/serviceAccounts/" + serviceAccountEmail, request)
           .execute();
 
-      System.out.println(
-          "Disabled service account: "
-              + "your-service-account-name@"
-              + projectId
-              + ".iam.gserviceaccount.com");
+      System.out.println("Disabled service account: " + serviceAccountEmail);
     } catch (IOException e) {
       System.out.println("Unable to disable service account: \n" + e.toString());
     }

--- a/iam/api-client/src/main/java/iam/snippets/EnableServiceAccount.java
+++ b/iam/api-client/src/main/java/iam/snippets/EnableServiceAccount.java
@@ -30,8 +30,9 @@ import java.util.Collections;
 public class EnableServiceAccount {
 
   // Enables a service account.
-  public static void enableServiceAccount(String projectId) {
+  public static void enableServiceAccount(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -41,24 +42,16 @@ public class EnableServiceAccount {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       EnableServiceAccountRequest request = new EnableServiceAccountRequest();
       service
           .projects()
           .serviceAccounts()
-          .enable(
-              "projects/-/serviceAccounts/"
-                  + "your-service-account-name@"
-                  + projectId
-                  + ".iam.gserviceaccount.com",
-              request)
+          .enable("projects/-/serviceAccounts/" + serviceAccountEmail, request)
           .execute();
 
-      System.out.println(
-          "Enabled service account: "
-              + "your-service-account-name@"
-              + projectId
-              + ".iam.gserviceaccount.com");
+      System.out.println("Enabled service account: " + serviceAccountEmail);
     } catch (IOException e) {
       System.out.println("Unable to enable service account: \n" + e.toString());
     }

--- a/iam/api-client/src/main/java/iam/snippets/ListServiceAccountKeys.java
+++ b/iam/api-client/src/main/java/iam/snippets/ListServiceAccountKeys.java
@@ -31,8 +31,9 @@ import java.util.List;
 public class ListServiceAccountKeys {
 
   // Lists all keys for a service account.
-  public static void listKeys(String projectId) {
+  public static void listKeys(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -42,17 +43,14 @@ public class ListServiceAccountKeys {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       List<ServiceAccountKey> keys =
           service
               .projects()
               .serviceAccounts()
               .keys()
-              .list(
-                  "projects/-/serviceAccounts/"
-                      + "your-service-account-name@"
-                      + projectId
-                      + ".iam.gserviceaccount.com")
+              .list("projects/-/serviceAccounts/" + serviceAccountEmail)
               .execute()
               .getKeys();
 

--- a/iam/api-client/src/main/java/iam/snippets/RenameServiceAccount.java
+++ b/iam/api-client/src/main/java/iam/snippets/RenameServiceAccount.java
@@ -30,8 +30,9 @@ import java.util.Collections;
 public class RenameServiceAccount {
 
   // Changes a service account's display name.
-  public static void renameServiceAccount(String projectId) {
+  public static void renameServiceAccount(String projectId, String serviceAccountName) {
     // String projectId = "my-project-id";
+    // String serviceAccountName = "my-service-account-name";
 
     Iam service = null;
     try {
@@ -41,17 +42,14 @@ public class RenameServiceAccount {
       return;
     }
 
+    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
     try {
       // First, get a service account using List() or Get()
       ServiceAccount serviceAccount =
           service
               .projects()
               .serviceAccounts()
-              .get(
-                  "projects/-/serviceAccounts/"
-                      + "your-service-account-name@"
-                      + projectId
-                      + ".iam.gserviceaccount.com")
+              .get("projects/-/serviceAccounts/" + serviceAccountEmail)
               .execute();
 
       // Then you can update the display name

--- a/iam/api-client/src/test/java/iam/snippets/ServiceAccountTests.java
+++ b/iam/api-client/src/test/java/iam/snippets/ServiceAccountTests.java
@@ -21,6 +21,7 @@ import static org.junit.Assert.assertNotNull;
 
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
+import java.util.UUID;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.BeforeClass;
@@ -36,6 +37,7 @@ public class ServiceAccountTests {
 
   private ByteArrayOutputStream bout;
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String SERVICE_ACCOUNT = "service-account-" + UUID.randomUUID();
 
   private static void requireEnvVar(String varName) {
     assertNotNull(
@@ -63,9 +65,9 @@ public class ServiceAccountTests {
 
   @Test
   public void stage1_testServiceAccountCreate() {
-    CreateServiceAccount.createServiceAccount(PROJECT_ID);
+    CreateServiceAccount.createServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
-    assertThat(got, containsString("Created service account: your-service-account-name"));
+    assertThat(got, containsString("Created service account: " + SERVICE_ACCOUNT));
   }
 
   @Test
@@ -77,49 +79,49 @@ public class ServiceAccountTests {
 
   @Test
   public void stage2_testServiceAccountRename() {
-    RenameServiceAccount.renameServiceAccount(PROJECT_ID);
+    RenameServiceAccount.renameServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Updated display name"));
   }
 
   @Test
   public void stage2_testServiceAccountKeyCreate() {
-    CreateServiceAccountKey.createKey(PROJECT_ID);
+    CreateServiceAccountKey.createKey(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Created key:"));
   }
 
   @Test
   public void stage2_testServiceAccountKeysList() {
-    ListServiceAccountKeys.listKeys(PROJECT_ID);
+    ListServiceAccountKeys.listKeys(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Key:"));
   }
 
   @Test
   public void stage3_testServiceAccountKeyDelete() {
-    DeleteServiceAccountKey.deleteKey(PROJECT_ID);
+    DeleteServiceAccountKey.deleteKey(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Deleted key:"));
   }
 
   @Test
   public void stage4_testDisableServiceAccount() {
-    DisableServiceAccount.disableServiceAccount(PROJECT_ID);
+    DisableServiceAccount.disableServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Disabled service account:"));
   }
 
   @Test
   public void stage5_testEnableServiceAccount() {
-    EnableServiceAccount.enableServiceAccount(PROJECT_ID);
+    EnableServiceAccount.enableServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Enabled service account:"));
   }
 
   @Test
   public void stage6_testServiceAccountDelete() {
-    DeleteServiceAccount.deleteServiceAccount(PROJECT_ID);
+    DeleteServiceAccount.deleteServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
     String got = bout.toString();
     assertThat(got, containsString("Deleted service account:"));
   }

--- a/iam/api-client/src/test/java/iam/snippets/ServiceAccountTests.java
+++ b/iam/api-client/src/test/java/iam/snippets/ServiceAccountTests.java
@@ -37,7 +37,8 @@ public class ServiceAccountTests {
 
   private ByteArrayOutputStream bout;
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
-  private static final String SERVICE_ACCOUNT = "service-account-" + UUID.randomUUID();
+  private static final String SERVICE_ACCOUNT =
+      "service-account-" + UUID.randomUUID().toString().substring(0, 8);
 
   private static void requireEnvVar(String varName) {
     assertNotNull(

